txpipe/twopoint_fourier.py

```diff
@@ -844,23 +863,45 @@ def save_power_spectra(self, tracers, nbin_source, nbin_lens):
                     i=d.i,
                     j=d.j,
                     n_ell=d.noise[i],
-                    bandpower_index=i,
+                    window_ind=i,
                 )
 
             # Add n_ell_coupled to tracer metadata. This will work as far as
             # the coupled noise is constant
             tr = S.tracers[tracer1]
             if (tracer1 == tracer2) and ("n_ell_coupled" not in tr.metadata):
-                if self.config["analytic_noise"] is False:
-                    # If computed through simulations, it might be better to
-                    # take the mean since, for now, only a float can be passed
-                    i = 0 if "lens" in tracer1 else 2
-                    tr.metadata["n_ell_coupled"] = np.mean(d.noise_coupled)
+
+                if self.config["gaussian_sims_factor"] != [1.] and "lens" in tracer1:
+                    
+                    print (tracer1)
+                    print("ATTENTION: We are multiplying the coupled noise saved in the sacc file by the gaussian sims factor.")
+                    print("Original noise:", d.noise_coupled)
+
+                    if self.config["analytic_noise"] is False:
+                        # If computed through simulations, it might be better to
+                        # take the mean since, for now, only a float can be passed
+                        i = 0 if "lens" in tracer1 else 2
+                        tr.metadata["n_ell_coupled"] = np.mean(d.noise_coupled*self.config["gaussian_sims_factor"][int(tracer1[-1])]**2)
+                    else:
+                        # Save the last element because the first one is zero for
+                        # shear
+                        tr.metadata["n_ell_coupled"] = d.noise_coupled[-1]*self.config["gaussian_sims_factor"][int(tracer1[-1])]**2
```

Comment:
This assumes you will have up to 10 lens tracers. But if you end up having 11, `tracer1[-1]` will be 1 instead of 10.

Reply:
I think that's OK now because it's only for the gaussian sims case where we have 5 bins only.